### ISSUE TITLE
Fix Request: AvoidFutureJoinWithoutTimeout false positives

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#307]
The following cases present false positives:
```
    public void getUserDetail4_OK(String userId) {
        CompletableFuture<Optional<String>> firstCompletableFuture = firstService.getFirst(userId);
        CompletableFuture<Optional<String>> secondCompletableFuture = secondService.getSecond(userId);

        CompletableFuture.allOf(firstCompletableFuture, secondCompletableFuture)
                .orTimeout(TIMEOUT_IN_MS, TimeUnit.MILLISECONDS)
                .join(); // good, but FALSE POSITIVE
    }

    private Orders getOrders_OK(OrderSearchOptions searchOptions, List<IGetOrdersService> getOrdersServices) {

        List<CompletableFuture<Object>> completableFutures = getOrdersServices.stream()
                .map(orderManager -> {
                    return CompletableFuture.supplyAsync(() -> i, executor).orTimeout(TIMEOUT_IN_MS, TimeUnit.MILLISECONDS);
                })
                .collect(Collectors.toList());

        List<Object> orderSearchResults = completableFutures.stream()
                .filter(future -> future.isDone() && !future.isCompletedExceptionally()) // keep only the ones completed -- added to deal with timeout
                .map(CompletableFuture::join) // good since timeout used above in method, but FALSE POSITIVE
                //.map(future -> future.join()) // alternative
                .collect(Collectors.toList());

        return new Orders(orderSearchResults);
    }
```
